tests/src/test_cases/test_file_explorer.c

```diff
@@ -4,42 +4,72 @@
 
 #include "unity/unity.h"
 #include <unistd.h>
+#include <stdio.h>
 #include <sys/stat.h>
+#include <fcntl.h>
 
 static lv_obj_t * active_screen = NULL;
 static lv_obj_t * file_explorer_obj;
 static lv_file_explorer_t * file_explorer;
 static lv_table_t * file_table;
 
+static void read_dir(void);
+
 void setUp(void)
 {
     active_screen = lv_screen_active();
     file_explorer_obj = lv_file_explorer_create(active_screen);
     file_explorer = (lv_file_explorer_t *)file_explorer_obj;
     file_table = (lv_table_t *)file_explorer->file_table;
+
+    mkdir("src/test_files/test_file_explorer_folder", 0777);
+    mkdir("src/test_files/test_file_explorer_folder/dev", 0777);
+    int fd = open("src/test_files/test_file_explorer_folder/video.mp4", O_CREAT);
+    close(fd);
+    fd = open("src/test_files/test_file_explorer_folder/audio.mp3", O_CREAT);
+    close(fd);
+    fd = open("src/test_files/test_file_explorer_folder/image.jpg", O_CREAT);
+    close(fd);
```

Comment:
```suggestion
    int fd = open("src/test_files/test_file_explorer_folder/video.mp4", O_CREAT, 0777);
    close(fd);
    fd = open("src/test_files/test_file_explorer_folder/audio.mp3", O_CREAT, 0777);
    close(fd);
    fd = open("src/test_files/test_file_explorer_folder/image.jpg", O_CREAT, 0777);
    close(fd);
```

`open` is a sort of variadic function that takes 2 or 3 parameters.

When the `flags` param of `open` includes `O_CREAT` or `O_TMPFILE`, then the third argument `mode` of `open` will be used. Since it's not supplied here, it's an undefined value - the value of some register or leftover value on the stack.

See `man 2 open`

> The  mode argument specifies the file mode bits be applied when
a new file is created.  This argument  must  be  supplied  when
O_CREAT  or O_TMPFILE is specified in flags

Reply:
Nice catch! Fixed and rebased